File: ncap/cap_core.py
"""
Packet capture core
"""
from scapy.all import conf, get_if_list
from .cap_device import CapDevice


class CapCore:
    """Packet capture core class"""

    def __init__(self):
        """Initialize capture core"""
        pass

    def get_device(self, device_name: str) -> str:
        """
        Get network device by name

        Args:
            device_name: Device description or name

        Returns:
            Device name

        Raises:
            ValueError: If device not found
        """
        # Get all available devices
        devices = get_if_list()

        # Check if device exists
        if device_name in devices:
            return device_name

        raise ValueError(f"网卡设备不存在: {device_name}")

    def start(self, device_name: str):
        """
        Start packet capture

        Args:
            device_name: Device to capture from

        Raises:
            ValueError: If device not found
        """
        device = self.get_device(device_name)
        cap_device = CapDevice(device, device_name)
        cap_device.start()

File: ncap/cap_core.go
package ncap

import (
	"fmt"

	"github.com/google/gopacket/pcap"
)

// CapCore packet capture core class
type CapCore struct{}

// NewCapCore creates a new packet capture core
func NewCapCore() *CapCore {
	return &CapCore{}
}

// GetDevice gets network adapter device
func (cc *CapCore) GetDevice(deviceName string) (*pcap.Handle, error) {
	// Find all network devices
	devices, err := pcap.FindAllDevs()
	if err != nil {
		return nil, fmt.Errorf("failed to get network adapter list: %v", err)
	}

	// Find device with specified name
	for _, device := range devices {
		if device.Description == deviceName {
			// Open device
			handle, err := pcap.OpenLive(
				device.Name,
				1024*1024*10,
				true,
				pcap.BlockForever,
			)
			if err != nil {
				return nil, fmt.Errorf("cannot open network adapter %s: %v", device.Name, err)
			}
			return handle, nil
		}
	}

	return nil, fmt.Errorf("network adapter device does not exist: %s", deviceName)
}

// Start begins packet capture
func (cc *CapCore) Start(deviceName string) error {
	device, err := cc.GetDevice(deviceName)
	if err != nil {
		return fmt.Errorf("failed to get network adapter: %v", err)
	}
	capDevice := NewCapDevice(device, deviceName)
	return capDevice.Start()
}
